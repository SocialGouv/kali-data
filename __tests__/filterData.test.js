import {
  latestArticleVersionFilter,
  isValidSection,
  filterData
} from "../src/filterData";

const sampleConvention = require("./kaliCont-sample.json");

const sampleArticles = [
  {
    id: "KALIARTI000001",
    cid: 1
  },
  {
    id: "KALIARTI000003",
    cid: 1
  },
  {
    id: "KALIARTI000002",
    cid: 1
  }
];

describe("latestArticleVersionFilter", () => {
  test("latestArticleVersionFilter should return false if its an old version of the article", () => {
    expect(
      latestArticleVersionFilter(
        {
          id: "KALIARTI000001",
          cid: 1
        },
        0,
        sampleArticles
      )
    ).toMatchSnapshot();
  });

  test("latestArticleVersionFilter should return true if its the newer version of the article", () => {
    expect(
      latestArticleVersionFilter(
        {
          id: "KALIARTI000003",
          cid: 1
        },
        0,
        sampleArticles
      )
    ).toMatchSnapshot();
  });
});

describe("isValidSection", () => {
  test("should exclude section ABROGE", () => {
    expect(isValidSection({ etat: "ABROGE" })).toEqual(false);
  });
  test("should exclude section PERIME", () => {
    expect(isValidSection({ etat: "PERIME" })).toEqual(false);
  });
  test("should include section VIGUEUR", () => {
    expect(isValidSection({ etat: "VIGUEUR" })).toEqual(true);
  });
  test("should include section VIGUEUR_ETEN", () => {
    expect(isValidSection({ etat: "VIGUEUR_ETEN" })).toEqual(true);
  });
  test("should include section VIGUEUR_NON_ETEN", () => {
    expect(isValidSection({ etat: "VIGUEUR_NON_ETEN" })).toEqual(true);
  });
  test("should include section without etaet", () => {
    expect(isValidSection({})).toEqual(true);
  });
});

describe("filterData", () => {
  test("should sort and filter sample convention", () => {
    expect(filterData(sampleConvention)).toMatchSnapshot();
  });
});
