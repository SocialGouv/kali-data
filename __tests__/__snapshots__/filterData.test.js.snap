// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterData should sort and filter sample convention 1`] = `
Object {
  "categorisation": Array [
    "categorie",
  ],
  "cid": undefined,
  "content": undefined,
  "etat": "VIGUEUR",
  "id": "1",
  "intOrdre": 1,
  "num": "292",
  "sections": Array [
    Object {
      "articles": Array [
        Object {
          "cid": "31",
          "content": undefined,
          "id": "3",
          "intOrdre": undefined,
          "num": undefined,
          "title": undefined,
        },
      ],
      "cid": undefined,
      "content": undefined,
      "etat": "VIGUEUR",
      "id": "KALISCTA3",
      "intOrdre": 1111,
      "num": undefined,
      "sections": Array [
        Object {
          "cid": undefined,
          "content": undefined,
          "etat": "VIGUEUR",
          "id": "KALISCTA31",
          "intOrdre": 31,
          "num": undefined,
          "sections": Array [
            Object {
              "articles": Array [
                Object {
                  "cid": "311",
                  "content": undefined,
                  "id": "311",
                  "intOrdre": undefined,
                  "num": undefined,
                  "title": undefined,
                },
              ],
              "cid": undefined,
              "content": undefined,
              "etat": "VIGUEUR",
              "id": "KALISCTA311",
              "intOrdre": 311,
              "num": undefined,
              "title": "title311",
            },
          ],
          "title": "title3",
        },
        Object {
          "articles": Array [
            Object {
              "cid": "321",
              "content": undefined,
              "id": "321",
              "intOrdre": undefined,
              "num": undefined,
              "title": undefined,
            },
          ],
          "cid": undefined,
          "content": undefined,
          "etat": "VIGUEUR",
          "id": "KALISCTA32",
          "intOrdre": 32,
          "num": undefined,
          "title": "title32",
        },
      ],
      "title": "title3",
    },
    Object {
      "cid": undefined,
      "content": undefined,
      "etat": "VIGUEUR",
      "id": "KALISCTA2",
      "intOrdre": 2222,
      "num": undefined,
      "sections": Array [
        Object {
          "articles": Array [
            Object {
              "cid": "211",
              "content": undefined,
              "id": "211",
              "intOrdre": undefined,
              "num": undefined,
              "title": undefined,
            },
            Object {
              "cid": "212",
              "content": undefined,
              "id": "212",
              "intOrdre": undefined,
              "num": undefined,
              "title": undefined,
            },
            Object {
              "cid": "213",
              "content": undefined,
              "id": "213",
              "intOrdre": undefined,
              "num": undefined,
              "title": undefined,
            },
          ],
          "cid": undefined,
          "content": undefined,
          "etat": "VIGUEUR",
          "id": "KALISCTA21",
          "intOrdre": 21,
          "num": undefined,
          "sections": Array [
            Object {
              "articles": Array [
                Object {
                  "cid": "2113",
                  "content": undefined,
                  "id": "2113",
                  "intOrdre": 7,
                  "num": undefined,
                  "title": undefined,
                },
                Object {
                  "cid": "2112",
                  "content": undefined,
                  "id": "2112",
                  "intOrdre": 55,
                  "num": undefined,
                  "title": undefined,
                },
                Object {
                  "cid": "2111",
                  "content": undefined,
                  "id": "2111",
                  "intOrdre": 555,
                  "num": undefined,
                  "title": undefined,
                },
              ],
              "cid": undefined,
              "content": undefined,
              "etat": "VIGUEUR",
              "id": "KALISCTA211",
              "intOrdre": 211,
              "num": undefined,
              "title": "title2",
            },
          ],
          "title": "title2-1",
        },
      ],
      "title": "title2",
    },
  ],
  "shortTitle": "shortTitle",
  "title": "title",
}
`;

exports[`latestArticleVersionFilter latestArticleVersionFilter should return false if its an old version of the article 1`] = `false`;

exports[`latestArticleVersionFilter latestArticleVersionFilter should return true if its the newer version of the article 1`] = `true`;
